### PR TITLE
feat: [CDS-39671]: help panel for pipeline steps

### PR DESCRIPTION
**Pull request by dhruvUpadhyay07**
### Summary

added help panels for pipeline steps 
added reference id as a property in steps class  , which can then be declared in respective steps to make use of the panel .

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 32 Issues 
* Code issue 32 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10443)

**Comment by dhruvUpadhyay07** [comment]
retrigger titlecheck

**Comment by dhruvUpadhyay07** [comment]
dockerize

**Review by kapilkumawat86 — APPROVED** [review]

File: src/modules/70-pipeline/components/PipelineStudio/RightDrawer/RightDrawer.tsx [hunk 1]
```diff
@@ -738,7 +740,12 @@ export function RightDrawer(): React.ReactElement {
     })
   }
 
+  const showHelpPanel = () => {
```
Comment (udayvunnam) [inline src/modules/70-pipeline/components/PipelineStudio/RightDrawer/RightDrawer.tsx:743]: nit: toggleHelpPanel.

showHelpPanel implies always showing the panel as `setHelpPanel(true)`

**Review by udayvunnam — APPROVED** [review]

**Pull request merged**